### DISCUSSION TITLE
Can we please have few examples to use chrome devtools

### DISCUSSION THREAD

**Discussion by radhakrishnanakireddy** [seleniumbase/SeleniumBase#1107] [General]
@mdmintz : Thanks a lot for providing support for selenium 4.
How we can use chrome devtools related methods in seleniumbase
can you  please help us with it

**Comment by mdmintz** [comment]
@radhakrishnanakireddy You can use the ``self.driver.execute_cdp_cmd(SCRIPT)`` command to execute methods specific to the Chrome Devtools Protocol. The official Selenium website has some documentation here: https://www.selenium.dev/documentation/support_packages/chrome_devtools/
I haven't tested any of that yet, but if you find any issues, it's on the Selenium Team. It looks like they are still actively developing changes to that. All ``self.driver.*`` commands go to the general Selenium WebDriver API, such as ``self.driver.execute_cdp_cmd(SCRIPT)``.

**Comment by mdmintz** [comment]
@radhakrishnanakireddy There are now a few examples that call ``self.driver.execute_cdp_cmd(SCRIPT)``:

* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/offline_examples/test_user_agent.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/user_agent_test.py

Both of those tests use ``execute_cdp_cmd()`` to change the browser's user agent string.
